lib/doc_my_routes/doc/hash_helpers.rb

```diff
@@ -0,0 +1,13 @@
+# Define error classes
+module DocMyRoutes
+  module HashHelpers
+    def self.deep_merge(first, second)
+      merger = proc { |key, f, s| f.is_a?(Hash) && s.is_a?(Hash) ? f.merge(s, &merger) : s }
```

Comment:
Can we name these variables oldval, newval or something other than single chars?